File: LAB3/app/src/main/java/com/example/tipper/BmrActivity.java
package com.example.tipper;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.RadioButton;
import android.widget.TextView;

public class BmrActivity extends AppCompatActivity {
    EditText height, weight, age;
    RadioButton sex_female, sex_male;
    TextView result;
    Button calculate, btnBack;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_bmr);

        height = (EditText) findViewById(R.id.height);
        weight = (EditText) findViewById(R.id.weight);
        age = (EditText) findViewById(R.id.age);
        sex_female = (RadioButton) findViewById(R.id.sex_female);
        sex_male = (RadioButton) findViewById(R.id.sex_male);
        result = (TextView) findViewById(R.id.result);
        calculate = (Button) findViewById(R.id.calculate);

        calculate.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                calculateBmr();
            }
        });

        btnBack = (Button) findViewById(R.id.btnBack);
        btnBack.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(BmrActivity.this, MainActivity.class);
                startActivity(intent);
            }
        });
    }

    private void calculateBmr() {

        double heightValue = Double.parseDouble(height.getText().toString());
        double weightValue = Double.parseDouble(weight.getText().toString());
        double ageValue = Double.parseDouble(age.getText().toString());
        String text = " Calories/day";
        if(sex_female.isChecked()) {

            double bmr = 447.593 + (9.247*weightValue) + (3.098*heightValue) - (4.330*ageValue);

            result.setText(bmr + text);
        }
        else if(sex_male.isChecked()) {

            double bmr = 88.362 + (13.397*weightValue) + (4.799*heightValue) - (5.677*ageValue);

            result.setText(bmr + text);
        }

    }


}

File: LAB4/app/src/main/java/com/example/myapplicationpamo/BmrActivity.kt
package com.example.myapplicationpamo

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.RadioButton
import android.widget.TextView

class BmrActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_bmr)

        val calculate = findViewById<Button>(R.id.calculate);
        val weight = findViewById<EditText>(R.id.weight);
        val height = findViewById<EditText>(R.id.height);
        val age = findViewById<EditText>(R.id.age);
        val sex_female = findViewById<RadioButton>(R.id.sex_female)
        val sex_male = findViewById<RadioButton>(R.id.sex_male)
        val result = findViewById<TextView>(R.id.result);

        calculate.setOnClickListener() {
            val input_height: String = height.text.toString()
            val value1: Double = input_height.toDouble()

            val input_weight: String = weight.text.toString()
            val value2: Double = input_weight.toDouble()

            val input_age: String = age.text.toString()
            val value3: Double = input_age.toDouble()

            if (sex_female.isChecked) {
                val bmrresult = femaleBmrCalculate(value1, value2, value3)
                result.text = "Calories/day " + "%.2f".format(bmrresult)
            } else if (sex_male.isChecked) {
                val bmrresult = maleBmrCalculate(value1, value2, value3)
                result.text = "Calories/day " + "%.2f".format(bmrresult)
            }

        }
    }

    private fun femaleBmrCalculate(v1: Double, v2: Double, v3: Double): Double {
        val bmrresult: Double = 447.593 + (9.247*v2) + (3.098*v1) - (4.330*v3)
        return bmrresult
    }

    private fun maleBmrCalculate(v1: Double, v2: Double, v3: Double): Double {
        val bmrresult: Double = 88.362 + (13.397*v2) + (4.799*v1) - (5.677*v3)
        return bmrresult
    }
}